### PR TITLE
Review section Implementation

### PR DESCRIPTION
**Pull request by clicelee**
### 📎 Title
Review section Implementation 

### 📃 Changes
- Review section

### 🫨 Considerations
order: image first and description follows
`img1` -> description1 -> `img2` -> description2

**1. About the card carousel…**
***first issue***
Screenshot 2024-07-22 at 10 54 05 PM

In Figma, the cards are displayed with width-full, 
Pasted Graphic
but the carousel I made displays only three cards at a time.

***second issue***
After dragging cards and hover them to see the description, I found this type of carousel issues. I think users will not feel comfortable having to click and drag just to move the cards. I don't know if it's an issue from shadcn UI or my code, but currently, dragging the card with the mouse doesn't feel very natural and not user friendly (plus, the hover effect is a bit distracting).
Screenshot 2024-07-22 at 11 15 11 PM

*So…* how about we try a different solution: using left and right button.🤔
If you think this is better, I'll send this issue to project managing team.

Screenshot 2024-07-22 at 10 44 28 PM

 (Additionally, I couldn't remove the buttons from the Shadcn carousel.😤  Once I remove the buttons, the next.js error occurs with BLANK ERROR MESSAGE. That's why I couldn't find the problem and leave the buttons. If we choose this solution, I won't need to remove the buttons anymore.)

**2. About the card description**
Screenshot 2024-07-22 at 10 38 37 PM
Screenshot 2024-07-22 at 10 39 33 PM

I tried to apply mb-[34px], but there’s a strange purple section in the card (you can check this with the Chrome Developer Tool), so the bottom margin is larger than I intended.
I need your help🥹

**3. About the font**
I haven’t applied the fonts yet. I’ll do that once the above issues are resolved.

### 🎇 Screenshots
Screenshot 2024-07-22 at 10 43 46 PM
Also you can check the screen recording video in the Notion 'review page'
The video was too large to upload 😅

### 💫 ETC
feel free to criticize my code

### CODE REVIEW

**Review by MinboyKim — COMMENTED** [review]
Well done!

## About the card carousel…

I also took a look at Carousel of shadcn/ui.

**Item width problem**
Basically, it seems like the width of the item is adjusted on a flex basis. So I tried giving `basis-auto` to Item and specifying width and height values ​​for Image seems to work well. I'm not sure if this is what you intended 🥲

**Drag and Hover problem**
It seems like dragging behavior was basically restricted, so I looked into it and found an option called `dragFree`. Setting the value to true seems to have made dragging more natural.
To improve the user experience and prevent the text from being accidentally selected while hovering over the Carousel, it would be good to add the `select-none` to the Carousel.

**Button problem**
I also tried deleting the buttons, and it worked fine. I just deleted `CarouselNext` and `CarouselPrevious` in `SlidingCard.tsx`. Can you share more accurate information about what  you tried?

**Good to add(Not essential)**
It seems that embla-carousel has an autoplay plugin. It can be added easily, so it would be a good idea to ask the PMs and add it.

Below is the code that applies all of the above.

```ts
// SlidingCard.tsx
'use client';

import Image from 'next/image';
import React from 'react';

import {
  Carousel,
  CarouselContent,
  CarouselPrevious,
} from '@/components/ui/carousel';
import { Card, CardContent } from '@/components/ui/card';
import AutoPlay from 'embla-carousel-autoplay';

...
...

export default function SlidingCard() {
  return (
    <Carousel
      opts={{
        align: 'start',
        dragFree: true,
      }}
      plugins={[
        AutoPlay({
          delay: 4000,
        }),
      ]}
      className="select-none"
    >
      <CarouselContent>
        {cards.map((card, index) => (
          <CarouselItem key={index} className="basis-auto">
            <div className="p-1">
              <Card className="group relative overflow-hidden">
                <CardContent className="p-0">
                  <Image
                    src={card.image}
                    alt={`Review Card ${index + 1}`}
                    className="w-[552px] h-[371px] rounded-[22px] transition-all duration-300 group-hover:brightness-50"
                  />
                  <div className="absolute inset-0 flex flex-col items-start opacity-0 group-hover:opacity-100 transition-opacity duration-300">
                    <h3 className="text-white text-[28px] font-semibold mt-[28px] ml-[34px]">
                      {card.title}
                    </h3>
                    <p className="text-white text-xl text-center p-[34px]">
                      {card.description}
                    </p>
                  </div>
                </CardContent>
              </Card>
            </div>
          </CarouselItem>
        ))}
      </CarouselContent>
    </Carousel>
  );
}
```

**result**
https://github.com/user-attachments/assets/6cb80347-2df6-41ce-a8d8-c9c5290f6142
(By default, autoplay stops after interaction. We can make it doesn't stop)

However, if you do not like it even after applying the above points, it would be good to go with the button method as you said. 👍

## About the card description
The purple section you mentioned appears to represent empty space. I think it would work well if you first make the contents of the div occupy the entire div and then give it some space.

Gooood job! 👏 I'll leave a comment for now and will approve once the work is completed.

File: app/_component/review/SlidingCard.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      <CarouselContent>
+        {cards.map((card, index) => (
+          <CarouselItem key={index} className="md:basis-1/2 lg:basis-1/3">
+            <div className="p-1">
+              <Card className="group relative overflow-hidden">
+              <CardContent className="p-0">
+                  <Image 
+                    src={card.image} 
+                    alt={`Review Card ${index + 1}`} 
+                    className="w-full h-auto rounded-[22px] transition-all duration-300 group-hover:brightness-50"
```
[the remark this answers is not in the document]
Comment (clicelee) [reply app/_component/review/SlidingCard.tsx:96]: Oh, I forgot that I had exported the image at double the size for better quality.😅
 I will add a class name to adjust the image size according to your advice.

File: app/_component/review/Review.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import SlidingCard from './SlidingCard.tsx';
```
[the remark this answers is not in the document]
Comment (clicelee) [reply app/_component/review/Review.tsx]: thanks I didn't know that
I accept the change

File: app/_component/review/Review.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+export default function Review() {
+  return (
+    <div className="h-screen w-screen flex justify-center items-center">
+      <div className="w-[80%}] text-center"> {/* text-center 클래스를 추가하여 중앙 정렬 */}
```
[the remark this answers is not in the document]
Comment (clicelee) [reply app/_component/review/Review.tsx:6]: Yesss.... 😅

**Comment by clicelee** [comment]
It's going to work well. Check it out and approve it please @MinboyKim

**Review by MinboyKim — APPROVED** [review]
Great job. LGTM 🐧

**Pull request merged**